algebird-core/src/main/scala/com/twitter/algebird/BinaryClassificationAUC.scala

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.algebird
+
+/**
+ * Curve is a list of Confusion Matrices with different
+ * thresholds
+ *
+ * @param matrices List of Matrices
+ */
+case class Curve(matrices: List[ConfusionMatrix])
+
+/**
+ * Given a List of (x,y) this functions computes the
+ * Area Under the Curve
+ */
+object AreaUnderCurve {
+  private def trapezoid(points: Seq[(Double, Double)]): Double = {
+    require(points.length == 2)
```

Comment:
do you want to take the two points directly instead of using this assertion, head, last combo?